docs/contributions.md

```diff
@@ -1,157 +1,200 @@
 
-#### Step 2: Branch
+### Step 2: Branch
 
 Create new branch in your forked copy of the `if` repository, which will contain your new feature, fix or change. 
 
-    ```bash
-    $ git checkout -b <topic-branch-name>
-    ```
+```bash
+$ git checkout -b <topic-branch-name>
+```
 
-#### Step 3: Commit
+### Step 3: Commit
 
 Make sure git knows your name and email address:
 
-   ```bash
-   $ git config --global user.name "Example User"
-   $ git config --global user.email "user@example.com"
-   ```
+```bash
+$ git config --global user.name "Example User"
+$ git config --global user.email "user@example.com"
+```
 
-Commiting multiple files with changes on multiple resources is not allowed.
+Each commit should cover one change to one resource. You should not add multiple changes to a single commit.
 Commit message should clearly describe on which resource changes are made.
+For the commit message, we adhere to the [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/) specification.
+Conventional commits are organized with a type, a scope and a description. The type can be one of: 
 
-Add and commit:
+- 'feat',
+- 'fix',
+- 'docs',
+- 'chore',
+- 'style',
+- 'refactor',
+- 'ci',
+- 'test',
+- 'revert',
+- 'package',
 
-   ```bash
-   $ git add my/changed/files
-   $ git commit
-   ```  
-    
-Commit your changes in logical chunks. Please do not push all changes in one commit.
+The scope is optional but should refer to the part of the codebase you are amending in the commit, e.g. `lib`, `types` etc.
 
-> Run `npm run fix` before commiting for not having conflict with CI linter.
+Here's an example of a valid commit message:
 
-Please adhere to these [Commit message guidelines](#commit-message-guidelines)
-   or your code is unlikely be merged into the main project.
+```
+feat(lib): initial commit for time-sync logic
+```
+
+or 
+
+```
+test(lib): in teads-curve add unit test to check that error is raised on missing tdp param 
+```
 
-#### Step 4: Sync
+Run `npm run fix` before commiting. If your commit message does not conform to the conventional commit specification or if you have not run `npm run fix` your commit will not satisfy the commitlint check.
 
-Use git pull/merge to synchronize your work with the IF repository.
+Add and commit with your commit message:
 
-   ```bash
-   $ git pull upstream dev
-   ```
-   
-#### Step 5: Push
+```bash
+$ git add my/changed/files
+$ git commit -m "<type-of-commit>(<my-optional-scope>): <my-commit-message>"
+```  
+
+### Step 5: Push
 
 Push your topic branch to your fork:
 
-   ```bash
-   $ git push origin <topic-branch-name>
-   ```
+```bash
+$ git push origin <topic-branch-name>
+```
 
-#### Step 6: Pull Request
+### Step 6: Pull Request
 
 Open a Pull Request from your fork of the repository to the `dev` branch of the IF repository with a clear title and description according to [template](.github/PULL_REQUEST_TEMPLATE.md).
```

Comment:
@jmcook1186 repository to the `dev` branch --> shouldn't it be `main` branch?

Reply:
```suggestion
Open a Pull Request from your fork of the repository to the `main` branch of the IF repository with a clear title and description according to [template](.github/PULL_REQUEST_TEMPLATE.md).
```